### ISSUE TITLE
Error after update HA

### ISSUE THREAD

**Issue by itsmartinho** [al-one/hass-xiaomi-miot#1903]
Hi, getting this error for 2 weeks or so:

> Logger: homeassistant.setup
> Source: setup.py:334
> First occurred: 10:55:34 PM (1 occurrences)
> Last logged: 10:55:34 PM
> 
> Setup failed for custom integration 'xiaomi_miot': Unable to import component: No module named 'imp'
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
>     component = await integration.async_get_component()
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 1027, in async_get_component
>     self._component_future.result()
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 1007, in async_get_component
>     comp = await self.hass.async_add_import_executor_job(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
>     result = self.fn(*self.args, **self.kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 1067, in _get_component
>     ComponentProtocol, importlib.import_module(self.pkg_path)
>                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
>     return func(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 995, in exec_module
>   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
>   File "/config/custom_components/xiaomi_miot/__init__.py", line 46, in <module>
>     from miio import (
>   File "/usr/local/lib/python3.12/site-packages/miio/__init__.py", line 12, in <module>
>     from miio.device import Device, DeviceStatus  # isort: skip
>     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.12/site-packages/miio/device.py", line 13, in <module>
>     from .miioprotocol import MiIOProtocol
>   File "/usr/local/lib/python3.12/site-packages/miio/miioprotocol.py", line 13, in <module>
>     import construct
>   File "/usr/local/lib/python3.12/site-packages/construct/__init__.py", line 22, in <module>
>     from construct.core import *
>   File "/usr/local/lib/python3.12/site-packages/construct/core.py", line 3, in <module>
>     import struct, io, binascii, itertools, collections, pickle, sys, os, tempfile, hashlib, importlib, imp
> ModuleNotFoundError: No module named 'imp'

What did I do wrong? This worked for years this way

**Comment by al-one** [comment]
- https://github.com/al-one/hass-xiaomi-miot/issues/1886#issuecomment-2409023449

**Comment by itsmartinho** [comment]
> * [Setup failed for custom integration 'xiaomi_miot': Unable to import component: No module named 'imp' #1886 (comment)](https://github.com/al-one/hass-xiaomi-miot/issues/1886#issuecomment-2409023449)

I appreciate it. I had removed that extention myself.
But still getting the same error. Do I need to update my python?

I've deleted that extention from HACS, restarted, and this is the logs I'm getting:

> Logger: homeassistant.setup
Source: setup.py:334
First occurred: 10:33:03 AM (1 occurrences)
Last logged: 10:33:03 AM

Setup failed for custom integration 'xiaomi_miot': Unable to import component: No module named 'imp'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 334, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1027, in async_get_component
    self._component_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1007, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1067, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 200, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/xiaomi_miot/__init__.py", line 46, in <module>
    from miio import (
  File "/usr/local/lib/python3.12/site-packages/miio/__init__.py", line 12, in <module>
    from miio.device import Device, DeviceStatus  # isort: skip
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/miio/device.py", line 13, in <module>
    from .miioprotocol import MiIOProtocol
  File "/usr/local/lib/python3.12/site-packages/miio/miioprotocol.py", line 13, in <module>
    import construct
  File "/usr/local/lib/python3.12/site-packages/construct/__init__.py", line 22, in <module>
    from construct.core import *
  File "/usr/local/lib/python3.12/site-packages/construct/core.py", line 3, in <module>
    import struct, io, binascii, itertools, collections, pickle, sys, os, tempfile, hashlib, importlib, imp
ModuleNotFoundError: No module named 'imp'

**Comment by dvbit** [comment]
Up.

**Comment by itsmartinho** [comment]
> Up.

Hi. 

I've managed to fix this.
I had ANOTHER app messing with this integration. It was HASS.agent for Windows. I had to install a another rep, and also removed the mentioned `[home-assistant-vacuum-styj02ym](https://github.com/KrzysztofHajdamowicz/home-assistant-vacuum-styj02ym).`.

Both were creating issues and the moment I removed them, everything else started working back again. 
Hope this helps.
